### ISSUE TITLE
DataLoader batching works not properly in mutations

### ISSUE THREAD

**Issue by livwvil** [ChilliCream/graphql-platform#7252]
### Product

Hot Chocolate

### Version

14.0.0-p.128

### Link to minimal reproduction

https://github.com/livwvil/HC_DataLoader_Issue/tree/master

### Steps to reproduce

1. Start the app
2. Send this request:
```graphql
query PersonByIdsWithDataLoader($personByIdsWithDataLoaderId: [ID!]!) {
  personByIdsWithDataLoader(id: $personByIdsWithDataLoaderId) {
    id
    mood
    friends {
      id
      mood
      friends {
        id
        mood
      }
    }      
  }
}
```
```
{
  "personByIdsWithDataLoaderId": ["UGVyc29uOgAAAAAAAAAAAAAAAAAAAAE=","UGVyc29uOgAAAAAAAAAAAAAAAAAAAAQ="]
}
```
3. Look at the console and check the batching `00000000-0000-0000-0000-000000000001` and `00000000-0000-0000-0000-000000000004`
4. Send this request:
```graphql
mutation MutPersonByIdsWithDataLoader($ids: [ID!]!) {
  mutPersonByIdsWithDataLoader(ids: $ids) {
    id
    mood
    friends {
      id
      mood
      friends {
        id
        mood
      }
    }    
  }
}
```
```
{
  "ids": ["UGVyc29uOgAAAAAAAAAAAAAAAAAAAAE=","UGVyc29uOgAAAAAAAAAAAAAAAAAAAAQ="]
}
```
5.  Look at the console and you won't see batching.

### What is expected?

Same behaviour for query and mutation. The other TestQueries and TestMutations methods work as expected.

### What is actually happening?

There is no batching in mutation. The code at line Test.cs:103 produces several requests through the dataLoader.

### Relevant log output

```shell
query PersonByIdsWithDataLoader:
PersonRepo.Get(00000000-0000-0000-0000-000000000001, 00000000-0000-0000-0000-000000000004)
PersonRepo.Get(00000000-0000-0000-0000-000000000002, 00000000-0000-0000-0000-000000000003, 00000000-0000-0000-0000-000000000005)
PersonRepo.Get(00000000-0000-0000-0000-000000000006)

mutation MutPersonByIdsWithDataLoader:
PersonRepo.Get(00000000-0000-0000-0000-000000000001)
PersonRepo.Get(00000000-0000-0000-0000-000000000004)
PersonRepo.Get(00000000-0000-0000-0000-000000000002, 00000000-0000-0000-0000-000000000003, 00000000-0000-0000-0000-000000000005)
PersonRepo.Get(00000000-0000-0000-0000-000000000006)
```

**Comment by michaelstaib** [comment]
Good catch and thank you for the detailed repro. The issue is with the execution engine which follows the spec to execute mutations serialized. The issue with that is, that we internally set the batch scheduler to dispatch on schedule when in a serialized context. This is a bug for mutations though. I have put it on out V14 backlog.

**Comment by michaelstaib** [comment]
This one is fixed now
